package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,228 @@
+    >>> cumulated_variance, principal_components = PSF_pca.fit()
+
+Inspect the components to determine the principal components you would like
+to retain. The choice is arbitrary, but I will stop when 95 percent of the
+variance is explained by the components.
+    >>> n_pcs = next(x[0] for x in enumerate(cumulated_variance) if x[1] > 0.95)
+    >>> pca_space = PSF_pca.transform(n_components=n_pcs)
+
+From here, inspection of the pca_space and conclusions to be drawn from the
+data are left to the user.
+"""
+from six.moves import range
+import logging
+import warnings
+
+import numpy as np
+from MDAnalysis import Universe
+from MDAnalysis.analysis.align import _fit_to
+
+from .base import AnalysisBase
+
+logger = logging.getLogger(__name__)
+
+
+class PCA(AnalysisBase):
+    """Principal component analysis on an MD trajectory
+
+    Attributes
+    ----------
+    p_components: array, (n_components, n_atoms)
+        The principal components of the feature space,
+        representing the directions of maximum variance in the data.
+    variance : array (n_components, )
+        The raw variance explained by each eigenvector of the covariance
+        matrix.
+    explained_variance : array, (n_components, )
+        Percentage of variance explained by each of the selected components.
+        If a subset of components is not chosen then all components are stored
+        and the sum of explained variances is equal to 1.0.
+
+    Methods
+    -------
+    fit(traj=None, select='All', start=None, stop=None, step=None)
+        Find the principal components of selected atoms in a subset of the
+        trajectory.
+    transform(traj=None, n_components=None)
+        Take the original trajectory and project it onto the principal
+        components.
+    inverse_tranform(pca_space)
+        Take a pca_space and map it back onto the trajectory used to create it.
+    """
+
+    def __init__(self, atomgroup, select='All', demean_ref=None,
+                 n_components=None, **kwargs):
+        """
+        Parameters
+        ----------
+        atomgroup: MDAnalysis atomgroup
+            AtomGroup to be used for PCA.
+        n_components : int, optional
+            The number of principal components to be saved, default saves
+            all principal components, Default: -1
+        start : int, optional
+            First frame of trajectory to use for generation
+            of covariance matrix, Default: None
+        stop : int, optional
+            Last frame of trajectory to use for generation
+            of covariance matrix, Default: None
+        step : int, optional
+            Step between frames of trajectory to use for generation
+            of covariance matrix, Default: None
+        """
+        super(PCA, self).__init__(atomgroup.universe.trajectory,
+                                  **kwargs)
+        self._u = atomgroup.universe
+        # for transform function
+        self.start = kwargs.get('start')
```

Comment:
the base init calls `_setup_frames` so they are set already at this point because you already called the base init function.